modules/axdma/src/allocator.rs

```diff
@@ -0,0 +1,76 @@
+use core::{alloc::Layout, ptr::NonNull};
+
+use allocator::{AllocError, AllocResult, BaseAllocator as _, ByteAllocator as _};
+use axalloc::global_allocator;
+use axhal::mem::virt_to_phys;
+use kspin::SpinNoIrq;
+use log::{debug, error};
+use memory_addr::{align_up, VirtAddr, PAGE_SIZE_4K};
+use os_dma::DMAInfo;
+use page_table_entry::MappingFlags;
+
+use crate::phys_to_bus;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "slab")] {
+        use allocator::SlabByteAllocator as DefaultByteAllocator;
```

Comment:
You can export `DefaultByteAllocator` in the `axalloc` and use them directly here to avoid adding features in this crate.